corehq/apps/products/views.py

```diff
@@ -113,51 +118,63 @@ def page_context(self):
 class FetchProductListView(ProductListView):
     urlname = 'commtrack_product_fetch'
 
-    def skip(self):
-        return (int(self.page) - 1) * int(self.limit)
-
-    def get_archive_text(self, is_archived):
-        if is_archived:
-            return _("This will re-activate the product, and the product will show up in reports again.")
-        return _("As a result of archiving, this product will no longer appear in reports. "
-                 "This action is reversable; you can reactivate this product by viewing "
-                 "Show Archived Products and clicking 'Unarchive'.")
-
     @property
     def product_data(self):
-        data = []
-        if self.show_inactive:
-            products = Product.archived_by_domain(
-                domain=self.domain,
-                limit=self.limit,
-                skip=self.skip(),
-            )
-        else:
-            products = Product.by_domain(
-                domain=self.domain,
-                limit=self.limit,
-                skip=self.skip(),
-            )
+        start = (self.page - 1) * self.limit
+        end = start + self.limit
+        return map(self.make_product_dict, self.product_queryset[start:end])
 
-        for p in products:
-            if p.program_id:
-                program = Program.get(p.program_id)
-            else:
-                program = get_or_create_default_program(self.domain)
-                p.program_id = program.get_id
-                p.save()
-
-            info = p._doc
-            info['program'] = program.name
-            info['edit_url'] = reverse('commtrack_product_edit', kwargs={'domain': self.domain, 'prod_id': p._id})
-            info['archive_action_desc'] = self.get_archive_text(self.show_inactive)
-            info['archive_action_text'] = _("Un-Archive") if self.show_inactive else _("Archive")
-            info['archive_url'] = reverse(
-                'unarchive_product' if self.show_inactive else 'archive_product',
-                kwargs={'domain': self.domain, 'prod_id': p._id}
-            )
-            data.append(info)
-        return data
```

Comment:
This method was starting to get big with the switch to explicitly specifying all the fields sent over the wire, so I pulled it apart.